src/istore.c

```diff
@@ -1055,7 +1053,59 @@ Datum istore_slice(PG_FUNCTION_ARGS)
         PG_RETURN_POINTER(result);
     }
     else
-        PG_RETURN_NULL();
+        PG_RETURN_EMPTY_ISTORE();
+}
+
+PG_FUNCTION_INFO_V1(istore_slice_min_max);
+Datum istore_slice_min_max(PG_FUNCTION_ARGS)
+{
+    int32   min = PG_GETARG_INT32(1);
+    int32   max = PG_GETARG_INT32(2);
+    IStore *is  = PG_GETARG_IS_COPY(0);
+
+    int min_idx = 0;
+    int i       = 0;
+    int len     = is->len;
+
+    IStorePair *pairs;
+    pairs = FIRST_PAIR(is, IStorePair);
+
+    if (min > max)
+        ereport(ERROR, (errcode(ERRCODE_SYNTAX_ERROR), errmsg("min must be less or equal max")));
+
+    if (is->len <= 0 || pairs[is->len - 1].key < min || pairs[0].key > max)
+        PG_RETURN_EMPTY_ISTORE();
+
+    if (pairs[0].key >= min && pairs[is->len - 1].key <= min)
```

Comment:
did you mean `pairs[is->len - 1].key <= max`?

Reply:
sure good catch